src/udp/socket.rs

```diff
@@ -1158,4 +1162,133 @@ mod test {
         });
         unwrap!(result)
     }
+
+    fn udp_rendezvous_connect_between_natted_hosts(start_delay: Duration) {
+        let _ = env_logger::init();
+
+        let mut core = unwrap!(Core::new());
+        let handle = core.handle();
+
+        let res = core.run(future::lazy(|| {
+            let (ch0, ch1) = util::two_way_channel();
+            let (server_drop_tx_0, server_drop_rx_0) = drop_notify();
+            let (server_drop_tx_1, server_drop_rx_1) = drop_notify();
+            let (server_addr_tx_0, server_addr_rx_0) = futures::sync::oneshot::channel();
+            let (server_addr_tx_1, server_addr_rx_1) = futures::sync::oneshot::channel();
+            let server_node = node::endpoint_v4(move |ip| {
+                let mut core = unwrap!(Core::new());
+                let handle = core.handle();
+
+                let res = core.run(future::lazy(move || {
+                    let server = unwrap!(UdpRendezvousServer::bind(&addr!("0.0.0.0:0"), &handle));
+                    let server_port = server.local_addr().port();
+                    let server_addr = SocketAddr::new(IpAddr::V4(ip), server_port);
```

Comment:
We can use `unspecified_to_localhost()` instead: https://github.com/ustulation/p2p/blob/master/src/socket_addr.rs#L34

Reply:
No we can't. The other peers aren't going to be able to connect to him over localhost - he's on the other side of the internet :)

Reply:
oh, right, my bad :)